admin/base/views.py

```diff
@@ -74,3 +57,15 @@ def form_valid(self, form):
     @property
     def success_url(self):
         raise NotImplementedError
+
+
+class GuidView(DetailView):
+    def __init__(self):
+        self.guid = None
+        super(GuidView, self).__init__()
+
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(GuidView, self).get(request, *args, **kwargs)
+        except AttributeError:
```

Comment:
This is where the class will actually return a page_not_found, either in this PR or in subsequent ones.